server/src/main/java/org/apache/druid/segment/realtime/appenderator/SinkQuerySegmentWalker.java

```diff
@@ -361,26 +359,48 @@ public <T> QueryRunner<T> getQueryRunnerForSegments(final Query<T> query, final
     }
   }
 
-  public void registerUpgradedPendingSegment(
-      SegmentIdWithShardSpec basePendingSegment,
-      SegmentIdWithShardSpec upgradedPendingSegment
-  )
+  /**
+   * Must be called when a segment is announced by a task.
+   * Either the base segment upon allocation or any upgraded version due to a concurrent replace
+   */
+  public void registerUpgradedPendingSegment(SegmentIdWithShardSpec id, Sink sink)
```

Comment:
These methods can be called simply `register`/`unregister` or `add`/`remove` since whether they are upgraded or not doesn't matter for this class.